include/ndcurves/polynomial_1d.h

```diff
@@ -0,0 +1,609 @@
+             const time_t max)
+      : dim_(1), degree_(1), t_min_(min), t_max_(max) {
+    if (t_min_ >= t_max_)
+      throw std::invalid_argument("T_min must be strictly lower than T_max");
+    coefficients_.setZero(2);
+    coefficients_ << init, (end - init) / (max - min);
+    safe_check();
+  }
+
+  ///
+  /// \brief Constructor from boundary condition with C1 :
+  /// create a polynomial that connect exactly init and end and thier first
+  /// order derivatives(order 3) \param init the initial point of the curve
+  /// \param d_init the initial value of the derivative of the curve
+  /// \param end the final point of the curve
+  /// \param d_end the final value of the derivative of the curve
+  /// \param min   : LOWER bound on interval definition of the spline.
+  /// \param max   : UPPER bound on interval definition of the spline.
+  ///
+  polynomial(const point_t& init, const point_t& d_init, const point_t& end,
+             const point_t& d_end, const time_t min, const time_t max)
+      : dim_(1), degree_(3), t_min_(min), t_max_(max) {
+    if (t_min_ >= t_max_)
+      throw std::invalid_argument("T_min must be strictly lower than T_max");
+    /** the coefficients [c0 c1 c2 c3] are found by solving the following system
+     * of equation (found from the boundary conditions) :
+     * [1  0  0   0   ] [c0]   [ init ]
+     * [1  T  T^2 T^3 ] [c1] = [ end  ]
+     * [0  1  0   0   ] [c2]   [d_init]
+     * [0  1  2T  3T^2] [c3]   [d_end ]
+     */
+    double T = max - min;
+    Eigen::Matrix<double, 4, 4> m;
+    m << 1., 0, 0, 0, 1., T, T * T, T * T * T, 0, 1., 0, 0, 0, 1., 2. * T,
+        3. * T * T;
+    Eigen::Matrix<double, 4, 4> m_inv = m.inverse();
+    Eigen::Matrix<double, 4, 1> bc;              // boundary condition vector
+    coefficients_ = coeff_t::Zero(degree_ + 1);  // init coefficient size
+    for (size_t i = 0; i < dim_;
+         ++i) {  // for each dimension, solve the boundary condition problem :
+      bc[0] = init;
+      bc[1] = end;
+      bc[2] = d_init;
+      bc[3] = d_end;
+      coefficients_.row(i) = (m_inv * bc).transpose();
+    }
+    safe_check();
+  }
+
+  ///
+  /// \brief Constructor from boundary condition with C2 :
+  /// create a polynomial that connect exactly init and end and thier first and
+  /// second order derivatives(order 5) \param init the initial point of the
+  /// curve \param d_init the initial value of the derivative of the curve
+  /// \param d_init the initial value of the second derivative of the curve
+  /// \param end the final point of the curve
+  /// \param d_end the final value of the derivative of the curve
+  /// \param d_end the final value of the second derivative of the curve
+  /// \param min   : LOWER bound on interval definition of the spline.
+  /// \param max   : UPPER bound on interval definition of the spline.
+  ///
+  polynomial(const point_t& init, const point_t& d_init, const point_t& dd_init,

```

Comment:
all these methods should not be member methods but free functions. These are effectively helper methods to create polynomials subject to constraints. in the general case they are not usuable and I don't think there is any justification for them to be member methods right?

Reply:
Well it's very needed for robotics application in general!
So we need the functionnality. Though we could juste provide the constructor with init pos,vel,acc and end pos,vel,acc. That would suffice.

Reply:
I believe this is the work of another PR.